src/utils/dexFormater.js

```diff
@@ -55,21 +60,33 @@ export const makeAssetToShow = (asset, symbol) => {
   return assetToShow;
 };
 
-export const formatProperlyValue = (asset, value, symbol, decimals) => {
+export const formatProperlyValue = (asset, value, decimals, symbol) => {
   let formattedValue;
-
   if (asset === 'Native') {
-    formattedValue = formatDollars(value);
+    formattedValue = formatDollars(value, true);
   } else if (asset === '1') {
-    formattedValue = formatMerits(value);
+    formattedValue = formatMerits(value, true);
   } else {
-    formattedValue = formatter(value, decimals);
+    formattedValue = formatAssets(value, decimals || 0, true);
   }
 
   const returnValue = symbol ? `${formattedValue} ${symbol}` : formattedValue;
   return returnValue;
 };
 
+export const parseProperlyValue = (asset, value, decimals) => {
+  let parsedValue;
+  if (asset === 'Native') {
+    parsedValue = parseDollars(value);
+  } else if (asset === '1') {
+    parsedValue = parseMerits(value);
+  } else {
+    parsedValue = parseAssets(value, decimals);
+  }
```

Comment:
If we have decimals, then we can just call `parseAssets(value, decimals)` even for LLD and LLM, which makes `parseProperlyValue` totally redundant.